java-frontend/src/main/java/org/sonar/java/JavaSquid.java

```diff
@@ -91,19 +94,28 @@ public JavaSquid(JavaVersion javaVersion, boolean xFileEnabled,
     astScannerForTests = new JavaAstScanner(sonarComponents);
     astScannerForTests.setVisitorBridge(createVisitorBridge(testCodeVisitors, testClasspath, javaVersion, sonarComponents, SymbolicExecutionMode.DISABLED));
 
+    //AstScanner for generated files
+    astScannerForGeneratedFiles = new JavaAstScanner(sonarComponents);
+    astScannerForGeneratedFiles.setVisitorBridge(createVisitorBridge(generatedCodeVisitors, classpath, javaVersion, sonarComponents, SymbolicExecutionMode.DISABLED));
   }
 
-  private static VisitorsBridge createVisitorBridge(
-    Iterable<JavaCheck> codeVisitors, List<File> classpath, JavaVersion javaVersion, @Nullable SonarComponents sonarComponents, SymbolicExecutionMode symbolicExecutionMode) {
+  private static VisitorsBridge createVisitorBridge(Iterable<? extends JavaCheck> codeVisitors, List<File> classpath,
+                                                    JavaVersion javaVersion, @Nullable SonarComponents sonarComponents,
+                                                    SymbolicExecutionMode symbolicExecutionMode) {
     VisitorsBridge visitorsBridge = new VisitorsBridge(codeVisitors, classpath, sonarComponents, symbolicExecutionMode);
     visitorsBridge.setJavaVersion(javaVersion);
     return visitorsBridge;
   }
 
 
   public void scan(Iterable<InputFile> sourceFiles, Iterable<InputFile> testFiles) {
```

Comment:
if this method is now supposed to be called only by tests, I would annotated it by `@VisibleForTests` and potentially change its visibility.

Reply:
I removed this overload, and instead added `emptyList()` parameter to the callers in tests.